### ISSUE TITLE
TrailingComma false positives in heredoc

### ISSUE THREAD

**Issue by abrom** [rubocop-hq/rubocop#6701]
## Expected behavior

I've found two cases where commas inside HEREDOC is tripping up the TrailingComma module. I've written spec tests that highlight the issue(s) below but not sure what the best solution is going to be. 

I'm wondering if a better solution for handling HEREDOC would be to strip it out or replace it with a regular string before trying to do the comma detection?

## Actual behavior

The specs below fail :(

## Steps to reproduce the problem

The following example specs fail:

In spec/rubocop/cop/style/trailing_comma_in_array_literal_spec.rb
```
expect_no_offenses(<<-RUBY.strip_indent)
  [
    <<-TEXT, 123
      Something with a , in it  # <-- Fails on this comma
    TEXT
  ]
RUBY
```
In this case it appears to be because the HEREDOC isn't the last parameter in the array

In spec/rubocop/cop/style/trailing_comma_in_arguments_spec.rb
```
expect_no_offenses(<<-RUBY.strip_indent)
  some_method(
    <<-LOREM.delete("\\n")
      Something with a , in it # <-- Fails on this comma
    LOREM
  )
RUBY
```
In this case, it seems to be the `delete("\n")` that is tripping up the HEREDOC detection

## RuboCop version

RuboCop 0.63.1  (current master)
https://github.com/rubocop-hq/rubocop/commit/42351e6d5ae73654a2318530a3fcc3861e94fa80
